src/ert/_c_wrappers/enkf/ert_workflow_list.py

```diff
@@ -91,112 +55,155 @@ def __init__(
-            c_ptr = self._alloc(subst_list, config_content, site_config_content)
+            content_dict = config_content_as_dict(config_content, site_config_content)
+
+            if ConfigKeys.LOAD_WORKFLOW_JOB in content_dict:
+                for workflow in content_dict.get(ConfigKeys.LOAD_WORKFLOW_JOB, []):
+                    _workflow_job_info.append(workflow)
+
+            if ConfigKeys.WORKFLOW_JOB_DIRECTORY in content_dict:
+                for workflow_dir in content_dict.get(
+                    ConfigKeys.WORKFLOW_JOB_DIRECTORY, []
+                ):
+                    _workflow_job_dir_info.append(workflow_dir)
+
+            if ConfigKeys.HOOK_WORKFLOW_KEY in content_dict:
+                for hook_name, mode_name in content_dict.get(
+                    ConfigKeys.HOOK_WORKFLOW_KEY, []
+                ):
+                    _hook_workflow_info.append([hook_name, mode_name])
+
+            if ConfigKeys.LOAD_WORKFLOW in content_dict:
+                for workflow in content_dict.get(ConfigKeys.LOAD_WORKFLOW, []):
+                    _workflow_info.append(workflow)
 
         if config_dict is not None:
-            workflow_joblist = WorkflowJoblist()
-            parser = WorkflowJob.configParser()
             for job in config_dict.get(ConfigKeys.LOAD_WORKFLOW_JOB, []):
-                try:
+                _workflow_job_info.append([job[ConfigKeys.PATH], job[ConfigKeys.NAME]])
+
+            for job_path in config_dict.get(ConfigKeys.WORKFLOW_JOB_DIRECTORY, []):
+                _workflow_job_dir_info.append(job_path)
+
+            if ConfigKeys.HOOK_WORKFLOW_KEY in config_dict:
+                for name, mode in config_dict.get(ConfigKeys.HOOK_WORKFLOW_KEY, []):
+                    _hook_workflow_info.append([name, mode])
+
+            for job in config_dict.get(ConfigKeys.LOAD_WORKFLOW, []):
+                _workflow_info.append([job[ConfigKeys.PATH], job[ConfigKeys.NAME]])
+
+        for workflow_job in _workflow_job_info:  # LOAD_WORKFLOW_JOB
+            try:
+                if len(workflow_job) == 1:
+                    new_job = WorkflowJob.fromFile(
+                        config_file=workflow_job[0],
+                        parser=parser,
+                    )
+                else:
                     new_job = WorkflowJob.fromFile(
-                        config_file=job[ConfigKeys.PATH],
-                        name=job[ConfigKeys.NAME],
+                        config_file=workflow_job[0],
+                        name=workflow_job[1],
                         parser=parser,
                     )
+            except OSError:
+                print(f"WARNING: Unable to create job from {workflow_job[0]}")
+                continue
+            if new_job is not None:
+                self._workflow_jobs[new_job.name()] = workflow_joblist.addJob(new_job)
+                new_job.convertToCReference(None)
+
+        for job_path in _workflow_job_dir_info:  # WORKFLOW_JOB_DIRECTORY
```

Comment:
just an annoying and unnecessary comment from my point of view - could it be removed?

Reply:
Sure. These are leftovers from when restructuring. Consider it done.